todo-app/src/component/TodosItem.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+
+//Component
+import EditForm from './EditForm';
+import DisplayForm from './DisplayForm';
+
+
+export default class TodosItem extends React.Component {
```

Comment:
A few points to make about this component.

First of all, you decided to put the "new todo" form state inside your MobX store as `newDescription` and `newDeadline`. Why did you decide to keep the editing values here as local React component state? You should choose one approach to form state in your application and stick to it instead of this inconsistency.

Second, you need to initialise all your state values inside your `state =` class property. You're setting state values here that aren't declared initially, and that leads to confusion for other developers and your future self.

Third, there is no reason to be setting state based on props in `componentDidMount`. You can do this directly inside your state class property.

Fourth, it appears that you are not passing the changed/saved values of your todo items into your store _at all_. You are only saving them in component local state. Take a look at your `handelSubmit` function. While it is sometimes acceptable to keep local state in form components, as soon as the data is "submitted" or "finalised", it needs to be integrated into the store.